smartapps/plaidsystems/spruce-scheduler.src/spruce-scheduler.groovy

```diff
@@ -147,39 +151,25 @@ def globalPage() {
         section('Push Notifications') {
                 input(name: 'notify', type: 'enum', title: 'Select what push notifications to receive.', required: false,
                     multiple: true, metadata: [values: ['Daily', 'Delays', 'Warnings', 'Weather', 'Moisture', 'Events']])
-                //obsolete input('recipients', 'contact', title: 'Send push notifications to', required: false, multiple: true)
                 input(name: 'logAll', type: 'bool', title: 'Log all notices to Hello Home?', defaultValue: 'false', options: ['true', 'false'])
         }
     }
 }
 
 def weatherPage() {
     dynamicPage(name: 'weatherPage', title: 'Weather settings') {
-       section('Location to get weather forecast and conditions:') {
-            href(name: 'hrefWithImage', title: "${zipString()}", page: 'zipcodePage',
-                 description: 'Set local weather station',
-                 required: false,
-                 image: 'http://www.plaidsystems.com/smartthings/rain.png'
-               )
+       section("Location to get weather forecast and conditions:") {
+            input(name: 'zipcode', type: 'text', title: "Zipcode.  Default is ${getPWSID()}", required: false, submitOnChange: true)
             input 'isRain', 'bool', title: 'Enable Rain check:', metadata: [values: ['true', 'false']]
             input 'rainDelay', 'decimal', title: 'inches of rain that will delay watering, default: 0.2', required: false
             input 'isSeason', 'bool', title: 'Enable Seasonal Weather Adjustment:', metadata: [values: ['true', 'false']]
         }
     }
 }
 
-def zipcodePage() {
-    return dynamicPage(name: 'zipcodePage', title: 'Spruce weather station setup') {
-        section(''){
-            input(name: 'zipcode', type: 'text', title: 'Zipcode. Default value is current Zip code',
-                defaultValue: getPWSID(), required: false, submitOnChange: true )
-        }
-    }
-}
-
 private String getPWSID() {
     String PWSID = location.zipCode
-    if (zipcode) PWSID = zipcode
+    if (!location.zipCode.isNumber()) PWSID = "${location.latitude.floatValue()},${location.longitude.floatValue()}"
```

Comment:
It might be best to do:

`!location.zipCode?.isNumber()` in case `location.zipCode` is null.

Reply:
I would even have checks on the latitude and longitude.